src/V3Param.cpp

```diff
@@ -961,15 +985,25 @@ class ParamVisitor final : public VNVisitor {
         // Must do ifaces first, so push to list and do in proper order
         string* const genHierNamep = new string(m_generateHierName);
         nodep->user5p(genHierNamep);
-        m_cellps.push_back(nodep);
+        // Visit parameters in the instantiation.
+        iterateChildren(nodep);
+        m_cellps.emplace(!VN_IS(nodep->modp(), Iface), nodep);
     }
 
     virtual void visit(AstClassRefDType* nodep) override {
-        if (nodep->paramsp()) {
-            nodep->paramsp()->v3warn(E_UNSUPPORTED, "Unsupported: parameterized classes");
-            pushDeletep(nodep->paramsp()->unlinkFrBackWithNext());
-        }
+        string* const genHierNamep = new string(m_generateHierName);
+        nodep->user5p(genHierNamep);
```

Comment:
```suggestion
        string* const genHierNamep = new string{m_generateHierName};
        nodep->user5p(genHierNamep);
```

use {} but I think would be better to make an e.g. setUser5pName function that takes a string and does the user5p assignment, as this code seems to now be several places.

Reply:
Done, moved to a new common function.